queries/idris/highlights.scm

```diff
@@ -0,0 +1,218 @@
+; ------------------------------------------------------------------------------
+; Literals and comments
+[
+  (integer)
+  (quantity)
+] @number
+
+(literal
+  (number)) @number.float
+
+(char) @character
+
+[
+  (string)
+  (pat_string)
+  (triple_quote_string)
+] @string
+
+(comment) @comment
+
+((comment) @comment.documentation
+  (#lua-match? @comment.documentation "^|||"))
+
+(unit) @constant
+
+; more general captures are moved to the top
+; before overwritten later by more specific captures
+(loname) @variable
+
+; ------------------------------------------------------------------------------
+; Punctuation
+[
+  "("
+  ")"
+  "{"
+  "@{"
+  "}"
+  "["
+  "[<"
+  "]"
+] @punctuation.bracket
+
+[
+  (comma)
+  (colon)
+  (pat_op)
+  (tuple_operator)
+] @punctuation.delimiter
+
+(pat_name
+  (loname) @variable.parameter)
+
+; ------------------------------------------------------------------------------
+; Types
+(signature
+  (loname) @type)
+
+; ------------------------------------------------------------------------------
+; Keywords, operators, imports
+[
+  "if"
+  "then"
+  "else"
+  "case"
+  "of"
+] @keyword.conditional
+
+(module) @module
+
+[
+  "import"
+  "module"
+  "namespace"
+  "parameters"
+] @keyword.import
+
+[
+  (operator)
+  (equal)
+  (wildcard)
+  "."
+  "|"
+  "=>"
+  "⇒"
+  "<="
+  "⇐"
+  "->"
+  "→"
+  "<-"
+  "←"
+  "\\"
+  "`"
+] @operator
+
+(qualified_loname
+  (caname) @module)
+
+(qualified_caname
+  (caname) @constructor)
+
+(qualified_operator
+  (caname) @module)
+
+(import
+  (caname) @module)
+
+[
+  (where)
+  "rewrite"
+  "interface"
+  "implementation"
+  "using"
+  "record"
+  "as"
+  "do"
+  (forall)
+  (fixity)
+  (impossible)
+  (with)
+  (proof)
+] @keyword.operator
+
+[
+  "data"
+  "let"
+  "in"
```

Comment:
What is the usage of these keywords? I don't know idris very well but I think "let" and "in" are typically just regular `@keyword`